src/accelerate/test_utils/test_script.py

```diff
@@ -108,6 +108,71 @@ def dl_preparation_check():
         print("Shuffled dataloader passing.")
 
 
+def central_dl_preparation_check():
+    state = AcceleratorState()
+    length = 32 * state.num_processes
+
+    dl = DataLoader(range(length), batch_size=8)
+    dl = prepare_data_loader(
+        dl, state.device, state.num_processes, state.process_index, put_on_device=True, central_dataloader=True
+    )
+    result = []
+    for batch in dl:
+        result.append(gather(batch))
+    result = torch.cat(result)
+    assert torch.equal(result.cpu(), torch.arange(0, length).long())
```

Comment:
Please add an error message if it fails :)

Reply:
Oh actually this applies to all `assert` in this file!

Reply:
This is just a test script ;-)